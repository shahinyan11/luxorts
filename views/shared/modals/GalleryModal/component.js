import { Modal } from 'antd';
import PropTypes from 'prop-types';
import { Swiper, SwiperSlide } from 'swiper/react';
import Image from 'next/image';

import SvgIcon from 'views/shared/SvgIcon';
import GradientButton from 'views/shared/GradientButton';

import useContainer from './hook';

const GalleryModal = ({ onClose, initialSlide, photos }) => {
  const { swiperRef, activeIndex, slidePrev, slideNext, onSlideChangeHandler } = useContainer({
    initialSlide,
  });

  return (
    <Modal
      visible
      className="modal-gallery"
      onCancel={onClose}
      closeIcon={<SvgIcon icon="modal-cross" className="modal__icon" />}
      footer={null}
    >
      <div className="modal__content">
        <p className="modal__text text-align-center mb-32 mb-lg-124">
          {`${activeIndex + 1} / `}
          <span className="modal__text-note">{photos.length}</span>
        </p>
        <Swiper ref={swiperRef} onSlideChange={onSlideChangeHandler}>
          {photos.map((photo) => (
            <SwiperSlide key={photo.id}>
              <div className="modal-gallery__slide-inner ml-auto mr-auto">
                <Image
                  blurDataURL={photo.photoUrls?.small || photo.src}
                  src={photo.photoUrls?.original || photo.src}
                  loading="eager"
                  layout="fill"
                  objectFit="contain"
                  placeholder="blur"
                  alt=""
                />
              </div>
              <p className="modal__text text-align-center mt-24 mb-0 mb-lg-112">
                {photo.description}
              </p>
            </SwiperSlide>
          ))}
          <GradientButton
            onClick={slidePrev}
            className="modal-gallery__nav modal-gallery__nav--prev main-btn--icon"
            variant="secondary"
            addonBefore={<SvgIcon icon="arrow-left" />}
            disabled={activeIndex === 0}
          />
          <GradientButton
            onClick={slideNext}
            className="modal-gallery__nav modal-gallery__nav--next main-btn--icon"
            variant="secondary"
            addonBefore={<SvgIcon icon="arrow-right" />}
            disabled={activeIndex === photos.length - 1}
          />
        </Swiper>
      </div>
    </Modal>
  );
};

GalleryModal.defaultProps = {
  onClose: undefined,
  initialSlide: 0,
};

GalleryModal.propTypes = {
  onClose: PropTypes.func,
  initialSlide: PropTypes.number,
  photos: PropTypes.arrayOf(PropTypes.shape()).isRequired,
};

export default GalleryModal;
