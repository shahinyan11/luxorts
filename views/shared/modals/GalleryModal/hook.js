import { useRef, useState } from 'react';

import useMount from 'hooks/useMount';
import useImmutableCallback from 'hooks/useImmutableCallback';

function useContainer({ initialSlide }) {
  const [activeIndex, setActiveIndex] = useState(initialSlide);
  const swiperRef = useRef();

  /**
   * Go to previous slide
   */
  const slidePrev = useImmutableCallback(() => {
    swiperRef.current.swiper.slidePrev();
  });

  /**
   * Go to next slide
   */
  const slideNext = useImmutableCallback(() => {
    swiperRef.current.swiper.slideNext();
  });

  /**
   * On slide change handler
   * @param swiper {object}
   */
  const onSlideChangeHandler = useImmutableCallback((swiper) => {
    setActiveIndex(swiper.activeIndex);
  });

  /**
   * On component mount handler
   */
  const onMountHandler = () => {
    swiperRef.current.swiper.slideTo(activeIndex);
  };

  /**
   * Lifecycle
   */
  useMount(onMountHandler);

  return {
    swiperRef,
    activeIndex,
    slidePrev,
    slideNext,
    onSlideChangeHandler,
    onMountHandler,
  };
}

export default useContainer;
