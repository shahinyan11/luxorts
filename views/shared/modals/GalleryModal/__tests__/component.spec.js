import { shallow } from 'enzyme';

import GalleryModal from '../component';

const mockedHookData = {
  swiperRef: { current: {} },
  activeIndex: 0,
  slidePrev: jest.fn(),
  slideNext: jest.fn(),
  onSlideChangeHandler: jest.fn(),
};
jest.mock('../hook', () => jest.fn(() => mockedHookData));

describe('GalleryModal component', () => {
  const props = {
    onClose: jest.fn(),
    initialSlide: 0,
    photos: [{ id: 'id', src: 'src', description: 'description' }],
  };
  const component = shallow(<GalleryModal {...props} />);

  it('matches snapshot', () => {
    expect(component).toMatchSnapshot();
  });
});
