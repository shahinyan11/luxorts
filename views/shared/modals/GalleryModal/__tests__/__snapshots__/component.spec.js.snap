// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GalleryModal component matches snapshot 1`] = `
<Modal
  className="modal-gallery"
  closeIcon={
    <SvgIcon
      className="modal__icon"
      icon="modal-cross"
    />
  }
  confirmLoading={false}
  footer={null}
  okType="primary"
  onCancel={[MockFunction]}
  visible={true}
  width={520}
>
  <div
    className="modal__content"
  >
    <p
      className="modal__text text-align-center mb-32 mb-lg-124"
    >
      1 / 
      <span
        className="modal__text-note"
      >
        1
      </span>
    </p>
    <Swiper
      onSlideChange={[MockFunction]}
    >
      <SwiperSlide
        key="id"
      >
        <div
          className="modal-gallery__slide-inner ml-auto mr-auto"
        >
          <Image
            alt=""
            blurDataURL="src"
            layout="fill"
            loading="eager"
            objectFit="contain"
            placeholder="blur"
            src="src"
          />
        </div>
        <p
          className="modal__text text-align-center mt-24 mb-0 mb-lg-112"
        >
          description
        </p>
      </SwiperSlide>
      <GradientButton
        addonAfter={null}
        addonBefore={
          <SvgIcon
            icon="arrow-left"
          />
        }
        className="modal-gallery__nav modal-gallery__nav--prev main-btn--icon"
        disabled={true}
        onClick={[MockFunction]}
        variant="secondary"
      />
      <GradientButton
        addonAfter={null}
        addonBefore={
          <SvgIcon
            icon="arrow-right"
          />
        }
        className="modal-gallery__nav modal-gallery__nav--next main-btn--icon"
        disabled={true}
        onClick={[MockFunction]}
        variant="secondary"
      />
    </Swiper>
  </div>
</Modal>
`;
